src/test/java/ro/msg/learning/shop/testcontainer/TestContainerTest.java

```diff
@@ -0,0 +1,54 @@
+package ro.msg.learning.shop.testcontainer;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import ro.msg.learning.shop.IntegrationTest;
+import ro.msg.learning.shop.model.Address;
+import ro.msg.learning.shop.model.Location;
+import ro.msg.learning.shop.repository.LocationRepository;
+
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+public class TestContainerTest extends IntegrationTest {
+
+    @Autowired
+    LocationRepository locationRepository;
```

Comment:
Again the repositories should not be used in tests

Reply:
I wanted to persist an entity in the database, so I used a repository. How should I modify it?

Reply:
If it is an integration test (which I see it is because it extends IntegrationTest), then you should call the endpoints using mockMvc.perform(). You already have a controller for testing purposes, you can create an endpoint there which will receive an entity as parameter and persist it in the DB.

Reply:
removed the class